### PR TITLE
Updated PhoneRepairShop_Code and PhoneRepairShop_Code.Tests files 

### PR DESCRIPTION
**Pull request by MehranQadri**
Updated PhoneRepairShop_Code files based on latest version of the customization project (build: 23-101-0069-0004). Updated PhoneRepairShop_Code.Tests file based on the one change made to it.

### CODE REVIEW

**Review by EAndrosova — APPROVED** [review]

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj [hunk 1]
```diff
@@ -35,9 +35,8 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="Autofac, Version=6.5.0.0, Culture=neutral, PublicKeyToken=17863af14b0044da, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\..\Bin\Autofac.dll</HintPath>
+    <Reference Include="Autofac, Version=7.0.0.0, Culture=neutral, PublicKeyToken=17863af14b0044da, processorArchitecture=MSIL">
+      <HintPath>..\packages\Autofac.7.0.0\lib\netstandard2.0\Autofac.dll</HintPath>
```
Comment (k-s-popova) [inline UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj]: @MehranQadri , don't we need to use Autofac library from the Bin folder?

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj [hunk 2]
```diff
@@ -49,26 +48,39 @@
       <HintPath>..\packages\Moq.4.14.1\lib\net45\Moq.dll</HintPath>
     </Reference>
     <Reference Include="PX.Common">
-      <HintPath>..\..\..\..\Bin\PX.Common.dll</HintPath>
+      <HintPath>..\..\..\..\..\PhoneRepairShopT280\Bin\PX.Common.dll</HintPath>
```
Comment (k-s-popova) [inline UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj]: Please revert this change. Same for others that include PhoneRepairShopT280 in the path

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj [hunk 3]
```diff
@@ -49,26 +48,39 @@
...
     <Reference Include="PX.Objects">
-      <HintPath>..\..\..\..\Bin\PX.Objects.dll</HintPath>
+      <HintPath>..\..\..\..\..\PhoneRepairShopT280\Bin\PX.Objects.dll</HintPath>
     </Reference>
     <Reference Include="PX.Tests.Unit">
-      <HintPath>..\..\..\..\Bin\PX.Tests.Unit.dll</HintPath>
+      <HintPath>..\..\..\..\..\PhoneRepairShopT280\App_Data\Projects\PhoneRepairShop\PX.Tests.Unit.dll</HintPath>
     </Reference>
     <Reference Include="System" />
+    <Reference Include="System.Buffers, Version=4.0.3.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
```
Comment (k-s-popova) [inline UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj]: If you include Autofac from the Bin folder, will these references still be added?

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/RSSVRepairPriceMaintTests.cs [hunk 4]
```diff
@@ -1,8 +1,9 @@
 ﻿using Xunit;
 using PX.Data;
-using PX.Objects.IN;
 using PX.Tests.Unit;
 using PhoneRepairShop;
+using PX.Objects.IN;
```
Comment (k-s-popova) [inline UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/RSSVRepairPriceMaintTests.cs]: Please try to not include the changes that do not add any meaning. Here you just change the order of lines.

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/RSSVWorkOrderEntryTests.cs [hunk 5]
```diff
@@ -116,17 +116,12 @@ public void TestRepairWorkOrdersForm()
             workOrder.ServiceID = repairService.ServiceID;
             graph.Caches[typeof(RSSVWorkOrder)].Update(workOrder);
 
-            Assert.Equal(2, graph.RepairItems.Select().Count);
-            Assert.Single(graph.Labor.Select());
```
Comment (k-s-popova) [inline UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/RSSVWorkOrderEntryTests.cs]: Is that expected that these asserts are removed?

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_code.sln [hunk 6]
```diff
@@ -2,50 +2,50 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio Version 17
 VisualStudioVersion = 17.3.32901.215
 MinimumVisualStudioVersion = 10.0.40219.1
-Project("{E24C65DC-7377-472B-9ABA-BC803B73C61A}") = "PhoneRepairShop", "..\..\..\", "{511A7860-082C-42D2-BB57-8FD0AD5F6C6F}"
+Project("{E24C65DC-7377-472B-9ABA-BC803B73C61A}") = "PhoneRepairShopT280-23-101-0069-0004", "..\..\..\", "{407B75C1-DF84-479F-BD54-F5A5D7012034}"
```
Comment (k-s-popova) [inline UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_code.sln]: PhoneRepairShop should not be replaced with PhoneRepairShopT280-23-101-0069-0004

File: UnitTesting/T280/PhoneRepairShop_Code/Solution.bat [hunk 7]
```diff
@@ -1 +1 @@
-"C:\Trainings\UpdateTCourses\App_Data\Projects\PhoneRepairShop_Code\PhoneRepairShop_Code.sln"
\ No newline at end of file
+"C:\T200SeriesCoursesInstances\PhoneRepairShopT280-23-101-0069-0004\App_Data\Projects\PhoneRepairShop\PhoneRepairShop_code.sln"
```
Comment (k-s-popova) [inline UnitTesting/T280/PhoneRepairShop_Code/Solution.bat]: Revert changes to this file.

**Review by k-s-popova — CHANGES_REQUESTED** [review]

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj (hunk 1 above)
Comment (MehranQadri) [reply UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj]: > @MehranQadri , don't we need to use Autofac library from the Bin folder?

Not necessarily. You can install it using Nuget package manager as I have done in this instance. Both options are valid and are mentioned in the text:
image

Comment (k-s-popova) [reply UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj]: Could you please confirm that both options are OK with an sme? Because in unit tests we probably use Autofac for dependency injection, and for dependency injection we have the recommendation to use the version from the Bin folder. (see https://help.acumatica.com/(W(149))/Help?ScreenId=ShowWiki&pageid=67164f06-37ff-40c1-9f33-55e6aa43b44e)

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/RSSVWorkOrderEntryTests.cs (hunk 5 above)
Comment (MehranQadri) [reply UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/RSSVWorkOrderEntryTests.cs]: Lines 116 and 117 got repeated somehow and 119 and 120 ended up  on a different line. I have fixed it.

File: UnitTesting/T280/PhoneRepairShop_Code/Solution.bat (hunk 7 above)
Comment (MehranQadri) [reply UnitTesting/T280/PhoneRepairShop_Code/Solution.bat]: done

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_code.sln (hunk 6 above)
Comment (MehranQadri) [reply UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_code.sln]: removed the T280-23-101-0069-0004 wherever applicable

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/RSSVRepairPriceMaintTests.cs (hunk 4 above)
Comment (MehranQadri) [reply UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/RSSVRepairPriceMaintTests.cs]: ok. I overwrote all the files in the Tests project with mine and hence these changes in the order happened in some files.

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj (hunk 3 above)
Comment (MehranQadri) [reply UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj]: Since I reverted back to the Autofac dll, I will remove these additional packages. They probably got installed alongside the Autofac package

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj (hunk 2 above)
Comment (MehranQadri) [reply UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj]: done.

File: UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj (hunk 1 above)
Comment (MehranQadri) [reply UnitTesting/T280/PhoneRepairShop_Code/PhoneRepairShop_Code.Tests/PhoneRepairShop_Code.Tests.csproj]: > Could you please confirm that both options are OK with an sme? Because in unit tests we probably use Autofac for dependency injection, and for dependency injection we have the recommendation to use the version from the Bin folder. (see https://help.acumatica.com/(W(149))/Help?ScreenId=ShowWiki&pageid=67164f06-37ff-40c1-9f33-55e6aa43b44e)

Thanks for sharing the article. Its best to stick with the Autofac dll provided in the acumatica instance in this case. I will revert this change and remove the addition of the autofac package.

**Comment by MehranQadri** [comment]
Actually I think it is best to revert my branch before all these changes. Messing with and reverting sln and csproj files can cause some serious issues. Since I did not change any thing in the PhoneRepairShop_Code.Tests project except for adding 1 using directive it is best to keep the original Tests project and just make that one change. I will only update the main PhoneRepairShop_Code project coz it has the latest changes and the one currentlly under UnitTesting\T280 is the old one.

**Review by k-s-popova — APPROVED** [review]
